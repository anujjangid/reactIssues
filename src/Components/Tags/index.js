// @flow

import React from 'react';
import { TagStyle } from '../styleComponent';

type Props = {
  data: [],
};
const Tags = ({ data }: Props) =>
  // $FlowFixMe
  data.map(tag => (
    <TagStyle key={tag.id} to={tag.url} color={tag.color}>
      {tag.name}
    </TagStyle>
  ));

export default Tags;
