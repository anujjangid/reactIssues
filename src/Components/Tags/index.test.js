import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Tags from "./index";

describe("Tags Components", () => {
  it("renders correctly without data", () => {
    const data = [];
    const tree = renderer
      .create(
        <BrowserRouter>
          <Tags data={data} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with data", () => {
    const data = [{ id: "Test", url: "test", color: "000" }];
    const tree = renderer
      .create(
        <BrowserRouter>
          <Tags data={data} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
