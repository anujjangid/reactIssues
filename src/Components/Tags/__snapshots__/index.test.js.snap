// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags Components renders correctly with data 1`] = `
<a
  className="sc-bZQynM bIjJNM"
  color="000"
  href="/test"
  onClick={[Function]}
/>
`;

exports[`Tags Components renders correctly without data 1`] = `null`;
